pkg/labels/labels.go

```diff
@@ -384,6 +384,16 @@ func NewLabelsFromSortedList(list string) Labels {
 	return NewLabelsFromModel(strings.Split(list, ";"))
 }
 
+// NewLabelArrayFromMap creates labels from a map
+func NewLabelArrayFromMap(base map[string]string, source string) LabelArray {
```

Comment:
This appears to be new functionality - why is this needed as part of this refactor? Factor this out into a separate commit and add additional unit tests for it.

Reply:
yup! It's actually only used in the follow-up PR (https://github.com/cilium/cilium/pull/8236/) but, in looking at this, I realised that it isn't directly needed. There are existing functions that can be composed to do it, albeit less efficiently. I'll go with that instead, since less code is better :)